api/v1beta1/azuremachine_types.go

```diff
@@ -66,8 +66,11 @@ type AzureMachineSpec struct {
 	// +optional
 	UserAssignedIdentities []UserAssignedIdentity `json:"userAssignedIdentities,omitempty"`
 
-	// RoleAssignmentName is the name of the role assignment to create for a system assigned identity. It can be any valid GUID.
-	// If not specified, a random GUID will be generated.
+	// SystemAssignedIdentityRole defines the role and scope to assign to the system assigned identity.
```

Comment:
Nit: it should really be "system-assigned identity" with a hyphen, although Azure itself is not consistent. No need to update, I just can't refrain from pointing this out.